tests/utils/test_libxc.py

```diff
@@ -0,0 +1,188 @@
+    (
+        'SCAN + rVV10',
+        'meta-GGA',
+        ['XC_MGGA_X_SCAN', 'XC_NLC_XC_RVV10', 'XC_NLCF_XC_RVV10'],
+    ),
+    (
+        'r2SCAN + VV10',
+        'meta-GGA',
+        ['XC_MGGA_X_R2SCAN', 'XC_NLC_XC_VV10', 'XC_NLCF_XC_VV10'],
+    ),
+    ('B97M-V (explicit)', 'meta-GGA', ['XC_MGGA_X_B97M_V', 'XC_NLC_XC_VV10']),
+    ('B97M-rV (explicit)', 'meta-GGA', ['XC_MGGA_X_B97M_RV', 'XC_NLC_XC_RVV10']),
+    ('TPSS + D3(BJ)', 'meta-GGA', ['XC_MGGA_X_TPSS', 'XC_MGGA_C_TPSS']),
+    ('SCAN + D3(BJ)', 'meta-GGA', ['XC_MGGA_X_SCAN', 'XC_MGGA_C_SCAN']),
+    ('r2SCAN + D4', 'meta-GGA', ['XC_MGGA_X_R2SCAN', 'XC_MGGA_C_R2SCAN']),
+    # ------------- GGA + NL combos --------
+    ('PBE + VV10', 'GGA', ['XC_GGA_X_PBE', 'XC_NLC_XC_VV10']),
+    ('PBE + rVV10', 'GGA', ['XC_GGA_X_PBE', 'XC_NLC_XC_RVV10']),
+    ('revPBE + VV10', 'GGA', ['XC_GGA_X_RPBE', 'XC_NLC_XC_VV10']),
+    ('BLYP + VV10', 'GGA', ['XC_GGA_X_B88', 'XC_GGA_C_LYP', 'XC_NLC_XC_VV10']),
+    ('PBE + D3(BJ)', 'GGA', ['XC_GGA_X_PBE', 'XC_GGA_C_PBE']),
+    ('PBEsol + D3', 'GGA', ['XC_GGA_X_PBE_SOL', 'XC_GGA_C_PBE_SOL']),
+    ('BP86 + D3', 'GGA', ['XC_GGA_X_B88', 'XC_GGA_C_P86']),
+    # ------------- Hybrids + NL -----------
+    ('B3LYP + D3(BJ)', 'hybrid-GGA', ['XC_HYB_GGA_XC_B3LYP']),
+    ('PBE0 + D3(BJ)', 'hybrid-GGA', ['XC_HYB_GGA_XC_PBE0', 'XC_HYB_GGA_XC_PBEH']),
+    ('HSE06 + D3(BJ)', 'hybrid-GGA', ['XC_HYB_GGA_XC_HSE06']),
+    (
+        'ωB97X-V (explicit)',
+        'hybrid-meta-GGA',
+        ['XC_HYB_MGGA_X_WB97X_V', 'XC_NLC_XC_VV10'],
+    ),
+    (
+        'ωB97M-V (explicit)',
+        'hybrid-meta-GGA',
+        ['XC_HYB_MGGA_X_WB97M_V', 'XC_NLC_XC_VV10'],
+    ),
+    # -------- a few “code nicknames” ------
+    ('PBE0 (alias PBEh)', 'hybrid-GGA', ['XC_HYB_GGA_XC_PBEH', 'XC_HYB_GGA_XC_PBE0']),
+    ('HSE (generic)', 'hybrid-GGA', ['XC_HYB_GGA_XC_HSE06', 'XC_HYB_GGA_XC_HSE03']),
+    ('B3PW', 'hybrid-GGA', ['XC_HYB_GGA_XC_B3PW91']),
+]
+
+
+@pytest.mark.parametrize('raw, expected_family, labels', _COMMON_XC_CASES)
+def test_dft_expands_100_common_functionals(raw, expected_family, labels):
+    dft = DFT()
+    dft.xc = XCFunctional(raw_string=raw)
+    dft.normalize(EntryArchive(), logger=logger)
+
+    assert dft.xc.components, f'Expansion produced no components for {raw}'
+    assert all(c.canonical_label for c in dft.xc.components)
+    assert dft.jacobs_ladder == expected_family
+
+    if expected_family.startswith('hybrid'):
+        # Schema must not infer α from names
+        assert dft.exact_exchange_mixing_factor is None
+
+
+@pytest.mark.parametrize('raw', ['PBE', 'B3LYP', 'HSE06', 'SCAN+rVV10'])
+def test_dft_expansion_is_idempotent_for_common_functionals(raw):
+    dft = DFT()
+    dft.xc = XCFunctional(raw_string=raw)
+    dft.normalize(EntryArchive(), logger=logger)
+    n1 = len(dft.xc.components or [])
+    dft.normalize(EntryArchive(), logger=logger)
+    n2 = len(dft.xc.components or [])
+    assert n2 == n1
```

Comment:
Isn't idempotency tested twice now? Btw, while it is a nice property, do you need this guarantee? I.e. are you ever running the expansion multiple times on the same hit?